lib/scrivener.ex

```diff
@@ -38,13 +38,12 @@ defmodule Scrivener do
   end
 
   @doc false
-  @spec paginate(any, Scrivener.Config.t()) :: Scrivener.Page.t()
+  @spec paginate(any, map | Keyword.t()) :: Scrivener.Page.t()
```

Comment:
Can you help me understand why this signature is changing and the one below is being removed?

Reply:
Sure. Having multiple signatures was producing an overlapping domain error with `mix dialyzer`:
```
lib/scrivener.ex:41:overlapping_contract
Overloaded contract for Scrivener.paginate/2 has
overlapping domains; such contracts are currently unsupported and
are simply ignored.
```

So we needed to condense down to just one `@spec` signature for the function. And since a `Scrivener.Config.t()` is also a `map`, this seemed the best way to characterize a common signature for both function clauses.

Reply:
Thanks for the explanation. Is it possible to keep the signature as `Scrivener.Config.t() | map | Keyword.t()`? I would like the type to specifically indicate that it accepts a `Scrivener.Config` if possible.

Reply:
Re-pushed with that change

Reply:
Pushed again with https://github.com/drewolson/scrivener/pull/57/commits/c29fc32ac8adfee593ca6b5ccf237a6fddc6438c, which stresses even more that want the arg to eventually be a `Config.t()`. I hope that's consistent with your ideas.